packages/core/src/sync-store/index.ts

```diff
@@ -725,55 +728,68 @@ export const createSyncStore = ({
         return { blockData, cursor };
       },
     ),
-  insertRpcRequestResult: async ({ request, blockNumber, chainId, result }) =>
-    database.wrap(
+  insertRpcRequestResults: async ({ requests, chainId }) => {
+    if (requests.length === 0) return;
+    return database.wrap(
       { method: "insertRpcRequestResult", includeTraceLogs: true },
       async () => {
+        const values = requests.map(({ request, blockNumber, result }) => ({
+          request_hash: crypto.createHash("md5").update(request).digest("hex"),
+          chain_id: chainId,
+          block_number: blockNumber,
+          result,
+        }));
+
         await database.qb.sync
           .insertInto("rpc_request_results")
-          .values({
-            request,
-            block_number: blockNumber,
-            chain_id: chainId,
-            result,
-          })
+          .values(values)
           .onConflict((oc) =>
-            oc.columns(["request_hash", "chain_id"]).doUpdateSet({ result }),
+            oc
+              .columns(["request_hash", "chain_id"])
+              .doUpdateSet({ result: sql`EXCLUDED.result` }),
           )
           .execute();
       },
-    ),
-  getRpcRequestResult: async ({ request, chainId }) =>
+    );
+  },
+  getRpcRequestResults: async ({ requests, chainId }) =>
     database.wrap(
-      { method: "getRpcRequestResult", includeTraceLogs: true },
+      { method: "getRpcRequestResults", includeTraceLogs: true },
       async () => {
+        const requestHashes = requests.map((request) =>
+          crypto.createHash("md5").update(request).digest("hex"),
```

Comment:
@typedarray Need to be very careful that this matches the previous hashes that we generated by postgres. I added a test for this.

Reply:
Seems good, but we should double check
- Make sure any pre-processing we're doing (casing, sorting keys?) is the same as before
- Casing, we're storing this in a TEXT column
- Any possibility that the strings were changed by the driver before making it to PG (in the old imp)

Reply:
The `request` is preprocessed before making it to this function.

https://github.com/ponder-sh/ponder/blob/689156e6878d62da3cdd7cb646a590ed68726ea8/packages/core/src/sync/transport.ts#L84-L86

Reply:
Any reason not to do the processing inside this function? Current pattern could make it easy for a caller to process incorrectly.